cmd/protoc-gen-multi/main.go

```diff
@@ -69,26 +70,24 @@ func generate(ctx context.Context, args []string, request *pluginpb.CodeGenerato
 	if err != nil {
 		return fmt.Errorf("invalid argument: %w", err)
 	}
-	for _, plugin := range plugins {
-		request.Parameter = proto.String(plugin.opt)
-		pluginInput, err := proto.Marshal(request)
-		if err != nil {
-			return err
-		}
+	pluginsN := len(plugins)
+	group, ctx := errgroup.WithContext(ctx)
+	outs := make([]pluginpb.CodeGeneratorResponse, pluginsN)
+	for i, plugin := range plugins {
+		plugin := plugin
+		pluginRequest := proto.Clone(request).(*pluginpb.CodeGeneratorRequest)
+		pluginResponse := &outs[i]
 		// Execute the plugin.
-		name := "protoc-gen-" + plugin.name
-		cmd := exec.CommandContext(ctx, name)
-		cmd.Stdin = bytes.NewReader(pluginInput)
-		var out bytes.Buffer
-		cmd.Stdout = &out
-		if err := cmd.Run(); err != nil {
-			return err
-		}
-		var pluginResponse pluginpb.CodeGeneratorResponse
-		if err := proto.Unmarshal(out.Bytes(), &pluginResponse); err != nil {
-			return err
-		}
+		group.Go(func() error {
+			return plugin.generate(ctx, pluginRequest, pluginResponse)
+		})
+	}
+	if err := group.Wait(); err != nil {
+		return err
+	}
+	for i := range outs {
```

Comment:
I guess that's one way to do it, another option would be to range over a channel gathering the code gen response.